dsew_community_profile/delphi_dsew_community_profile/pull.py

```diff
@@ -261,13 +320,13 @@ def download_and_parse(listing, logger):
 
 def nation_from_state(df, sig, geomapper):
     """Compute nation level from state df."""
-    if SIGNALS[sig]: # true if sig is a rate
-        df = geomapper.add_population_column(df, "state_code") \
+    if SIGNALS[sig]["is_rate"]: # true if sig is a rate
+        df = geomapper.add_population_column(df, "state_id") \
                       .rename(columns={"population":"weight"})
-        df.weight = df.weight / df.weight.sum()
+        df.weight = df.weight / df.weight.sum() * len(df.timestamp.unique())
```

Comment:
Is this still needed?

Reply:
With regards to the issue caused by multiple publish dates? Yes, this was a separate issue causing incorrect nation "positivity" values. When processing multiple days of data at a time, just summing the weights makes the normalization denominator too big. Then the normalized weights and the resulting nation values are too small.

Reply:
wouldn't we want to groupby timestamp and then sum in that case, instead of multiplying uniformly? or are we always guaranteed the same number of rows per unique timestamp?

Reply:
I'll switch to a groupby approach to be more robust.